src/app/core/user/user-account/user-account.service.ts

```diff
@@ -1,72 +1,59 @@
 import { Injectable } from "@angular/core";
 import { HttpClient, HttpHeaders } from "@angular/common/http";
-import { EntityMapperService } from "../../entity/entity-mapper.service";
-import { User } from "../user";
 
 @Injectable({
   providedIn: "root",
 })
 export class UserAccountService {
   private static readonly COUCHDB_USER_ENDPOINT = "/db/_users/org.couchdb.user";
-  constructor(
-    private http: HttpClient,
-    private entityMapper: EntityMapperService
-  ) {}
+  constructor(private http: HttpClient) {}
 
   /**
    * Function to change the password of a user
-   * @param user The user for which the password should be changed
+   * @param username The username for which the password should be changed
    * @param oldPassword The current plaintext password of the user
    * @param newPassword The new plaintext password of the user
    * @return Promise that resolves once the password is changed in _user and the database
    */
   public async changePassword(
-    user: User,
+    username: string,
     oldPassword: string,
     newPassword: string
-  ): Promise<User> {
-    if (!user.checkPassword(oldPassword)) {
-      throw new Error("Wrong current password");
-    }
-
+  ): Promise<void> {
     let userResponse;
     try {
-      userResponse = await this.getCouchDBUser(user, oldPassword);
+      // TODO due to cookie-auth, the old password is actually not checked
+      userResponse = await this.getCouchDBUser(username, oldPassword);
     } catch (e) {
       throw new Error("Current password incorrect or server not available");
     }
 
     userResponse["password"] = newPassword;
-    user.setNewPassword(newPassword);
     try {
-      await Promise.all([
-        this.saveNewPasswordToCouchDB(user, oldPassword, userResponse),
-        this.entityMapper.save<User>(user),
-      ]);
+      await this.saveNewPasswordToCouchDB(username, oldPassword, userResponse);
```

Comment:
shouldn't this also update the local storage?

Reply:
I was thinking about that too. But I didn't want to make the assumption that there is a `LocalSession` because the password change also works if only the `RemoteSession` is used where nothing should be written to the local storage. In this case, the `UserAccountComponent` executes a login after successful password change (see [here](https://github.com/Aam-Digital/ndb-core/pull/941/files#diff-662c352c412ac6bcc03b47a13ef7fcfee16b1c324141c7e7fb402cb75a5bbb2fR108)) which will then write the new credentials to the local storage